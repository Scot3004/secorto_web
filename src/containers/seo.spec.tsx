import * as React from "react";
import renderer from "react-test-renderer"
import Seo from "./seo"
import * as Gatsby from "gatsby"
import { ThemeProvider } from "theme-ui"

const useStaticQuery = jest.spyOn(Gatsby, `useStaticQuery`)
const mockUseStaticQuery = {
  site: {
    siteMetadata: {
      title: `Site metadata title`,
      siteUrl: 'https://example.site.metadata/',
      description: "Site metadata description"
    }
  }
}

const theme = {
  colors: {
    primary: "#fff",
  },
}

describe("Seo con titulo", () => {
  beforeEach(() => {
    useStaticQuery.mockImplementation(() => mockUseStaticQuery)
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it("render Seo modelo blog", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Seo title="Blog post" imageSource="imageSourceStub" imageAlt="Stub para el alt de imagen" description="description stub">
            <meta property="og:article:published_time" content="August 14, 2022"></meta>
          </Seo>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
