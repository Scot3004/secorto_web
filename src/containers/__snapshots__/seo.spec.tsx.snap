// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seo con titulo render Seo modelo blog 1`] = `
[
  <title>
    Blog post | Site metadata title
  </title>,
  <meta
    content="description stub"
    name="description"
  />,
  <meta
    content="Blog post"
    property="og:title"
  />,
  <meta
    content="description stub"
    property="og:description"
  />,
  <meta
    content="website"
    property="og:type"
  />,
  <meta
    content="Stub para el alt de imagen"
    property="og:image:alt"
  />,
  <meta
    content="summary"
    name="twitter:card"
  />,
  <meta
    content=""
    name="twitter:creator"
  />,
  <meta
    content="Blog post"
    name="twitter:title"
  />,
  <meta
    content="description stub"
    name="twitter:description"
  />,
  <meta
    content="Stub para el alt de imagen"
    property="twitter:image:alt"
  />,
  <meta
    content="https://example.site.metadata/imageSourceStub"
    property="og:image"
  />,
  <meta
    content="https://example.site.metadata/imageSourceStub"
    property="twitter:image"
  />,
  <meta
    content="width=device-width, minimum-scale=1"
    name="viewport"
  />,
  <meta
    content="August 14, 2022"
    property="og:article:published_time"
  />,
]
`;
